controllers/imagerepository_controller.go

```diff
@@ -100,23 +103,30 @@ func (r *ImageRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	patcher, err := patch.NewHelper(&imageRepo, r.Client)
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
+	defer func() {
+		if err := patcher.Patch(ctx, &imageRepo, patch.WithOwnedConditions{
+			Conditions: []string{meta.ReadyCondition},
+		}, patch.WithStatusObservedGeneration{}); err != nil {
+			retErr = kerrors.NewAggregate([]error{retErr, err})
+		}
+	}()
+
 	defer r.recordSuspension(ctx, imageRepo)
 
 	log := logr.FromContext(ctx)
 
 	if imageRepo.Spec.Suspend {
```

Comment:
In the source-controller we do not push any updates to the object in case it has been suspended (see: https://github.com/fluxcd/source-controller/blob/reconcilers-dev/controllers/bucket_controller.go#L95-L102).

Making this visual to the end-user should instead be done by including the `Suspend` as a column in the Custom Resource Definition, so that the value on the object is always reflected to table-views, etc.

Reply:
What's the rationale for not accounting for suspend in the status? I guess you can argue that one both ways -- on the one hand, third party libs aren't going to know about `.spec.suspend`, for instance.

Reply:
_goes looking for the previous run of this conversation_

Reply:
Suspend tells the reconciler to stop looking at the resource and pause all observations, if you reflect this, you are not pausing observations.

Reply:
That's how you've defined it, but why is that better than e.g., recording that it's not ready because it's suspended?

Reply:
Because continuing to push updates to the object will also continue triggering any watcher that is subscribed to the resource, it will continue bumping the `observedGeneration` on at least the Condition type, and probably some other adverse affects in the machinery area.

It is not just us who uses this approach: https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/machineset_controller.go#L132-L136

Reply:
Cluster API has a pause button so they can implement the `move` command, where they need to stop acting on the resources to be moved because they will be duplicated elsewhere, without upsetting other controllers. I don't think there's an analogue of that requirement for GOTK, but I can probably come up with other scenarios where that behaviour 
 is useful, specifically the "without upsetting other controllers" part.

Reply:
In other words, I'll remove that code if it's part of the GOTK contract.